bindings/python/src/data_file.rs

```diff
@@ -143,8 +143,8 @@ impl PyDataFile {
     }
 
     #[getter]
-    fn split_offsets(&self) -> &[i64] {
-        self.inner.split_offsets()
+    fn split_offsets(&self) -> Option<Vec<i64>> {
```

Comment:
```suggestion
    fn split_offsets(&self) -> Option<&[i64]> {
```

We don't need to collect into vector.